### PR TITLE
Bump to depend on iree 3.0.0 packages.

### PR DESCRIPTION
**Pull request by ScottTodd**
See the release tracker at https://github.com/iree-org/iree/issues/19063.

### CODE REVIEW

**Comment by marbre** [comment]
Seems there are some failures which might be due to the update :/

**Comment by ScottTodd** [comment]
> Seems there are some failures which might be due to the update :/

Ack. Fixed a few but looks like there are more.

File: iree/turbine/kernel/compiler/dispatch_codegen.py [hunk 1]
```diff
@@ -165,8 +165,8 @@ def abi_type(binding: BindingDesc):
                 def_func_args = list(def_func_block.arguments)
                 if workgroup_size is not None and subgroup_size is not None:
                     def_func_op.attributes["translation_info"] = Attribute.parse(
-                        f"#iree_codegen.translation_info<None "
-                        f"workgroup_size=[{','.join(str(x) for x in workgroup_size)}]"
+                        f"#iree_codegen.translation_info<pipeline = None "
+                        f"workgroup_size=[{','.join(str(x) for x in workgroup_size)}] "
                         f"subgroup_size={subgroup_size}>"
```
Comment (ScottTodd) [inline iree/turbine/kernel/compiler/dispatch_codegen.py:170]: @qedawkins @kuhar I'm still getting errors here in some tests. Not sure what I'm doing wrong

https://github.com/iree-org/iree-turbine/actions/runs/11861587460/job/33059220943?pr=275#step:6:42863

```
            with InsertionPoint(self.def_module.body_block):
                def_func_op = func_d.FuncOp(name, def_ftype)
                def_func_block = def_func_op.add_entry_block()
                def_func_args = list(def_func_block.arguments)
                if workgroup_size is not None and subgroup_size is not None:
>                   def_func_op.attributes["translation_info"] = Attribute.parse(
                        f"#iree_codegen.translation_info<pipeline = None "
                        f"workgroup_size=[{','.join(str(x) for x in workgroup_size)}] "
                        f"subgroup_size={subgroup_size}>"
                    )
E                   iree.compiler._mlir_libs._site_initialize.<locals>.MLIRError: Unable to parse attribute:
E                   error: "#iree_codegen.translation_info<pipeline = None workgroup_size=[128,2,1] subgroup_size=64>":1:32: expected ::mlir::iree_compiler::IREE::Codegen::DispatchLoweringPassPipeline to be one of: CPUDefault, CPUDoubleTilingExpert, CPUConvTileAndDecomposeExpert, Mmt4dTilingExpert, CPUBufferOpsTileAndVectorize, CPUDataTiling, CPULinalgExtTileAndVectorize, LLVMGPUDefault, LLVMGPUBaseLowering, LLVMGPUDistribute, LLVMGPUVectorize, LLVMGPUMatmulSimt, LLVMGPUMatmulTensorCore, LLVMGPUTransposeSharedMem, LLVMGPUWarpReduction, LLVMGPUPackUnPack, LLVMGPUMatmulTensorCoreMmaSync, LLVMGPUVectorDistribute, LLVMGPUPadAndVectorDistribute, LLVMGPUWinogradVectorize, LLVMGPUTileAndFuse, SPIRVBaseLowering, SPIRVBaseDistribute, SPIRVBaseVectorize, SPIRVSubgroupReduce, SPIRVMatmulPromoteVectorize, SPIRVCooperativeMatrixVectorize, SPIRVWinogradVectorize, VMVXDefault, TransformDialectCodegen, Custom, None
E                   error: "#iree_codegen.translation_info<pipeline = None workgroup_size=[128,2,1] subgroup_size=64>":1:41: failed to parse DispatchLoweringPassPipelineAttr parameter 'value' which is to be a `::mlir::iree_compiler::IREE::Codegen::DispatchLoweringPassPipeline`
E                   error: "#iree_codegen.translation_info<pipeline = None workgroup_size=[128,2,1] subgroup_size=64>":1:41: failed to parse IREECodegen_TranslationInfoAttr parameter 'passPipeline' which is to be a `IREE::Codegen::DispatchLoweringPassPipelineAttr`

iree/turbine/kernel/compiler/dispatch_codegen.py:167: MLIRError
```

Comment (ScottTodd) [reply iree/turbine/kernel/compiler/dispatch_codegen.py]: Ah, I think those workflows still installed the older versions. Failures should go away after pushing a stable release, or I can get them installing the nightly releases too.

Comment (ScottTodd) [reply iree/turbine/kernel/compiler/dispatch_codegen.py:170]: Passing `--upgrade` in the workflows helped. Looks like some of the runners are persistent and don't clear our their venvs between runs. FYI @harsh-nod

Comment (kuhar) [reply iree/turbine/kernel/compiler/dispatch_codegen.py:170]: Looks good in the current form

**Comment by ScottTodd** [comment]
Updated to latest stable IREE 3.0.0. PTAL?

**Comment by marbre** [comment]
I can confirm that the CI on main pulls in the iree-base-{compiler,runtime} packages with version 3.0.0. However, I also noticed https://github.com/iree-org/iree-turbine/actions/runs/11898714760/job/33155792634#step:5:86. Not necessarily something we need to fix now (as a release is manually build) but seems worth to at least fill an issue. WDYT?

**Comment by marbre** [comment]
Okay, let's handle it in a follow up. I assume we want to push nightly releases via GH anyway?

**Comment by ScottTodd** [comment]
Yeah we can start publishing nightly packages via GH. I'll file an issue in IREE to scrape other sources when building the index at https://iree.dev/pip-release-links.html, so we can have a single `--find-links --pre` that gets all of our packages.

**Comment by ScottTodd** [comment]
Filed https://github.com/iree-org/iree/issues/19193 for the consolidated nightly release index page

**Comment by ScottTodd** [comment]
Fixing the `--package-version 2.5.0.dev` in https://github.com/iree-org/iree-turbine/pull/314

**Pull request merged**